data_structures/src/transaction.rs

```diff
@@ -253,11 +253,31 @@ impl TxInclusionProof {
         }
     }
 
+    // Create a TX inclusion proof assuming the inputs are already Hashes
+    pub fn new_with_hashes(index: usize, leaves: Vec<Hash>) -> TxInclusionProof {
```

Comment:
This is basically the same as `TxInclusionProof::new`, so if you make it generic you will be able to implement `new` as a wrapper of `new_with_hashes`:

```suggestion
    pub fn new_with_hashes<I: IntoIterator<Item = Hash>>(
        index: usize,
        leaves: I,
    ) -> TxInclusionProof {
```

Then `TxInclusionProof::new` becomes one line:

```rust
        Self::new_with_hashes(index, leaves.into_iter().map(|t| t.hash()))
```